virttest/utils_misc.py

```diff
@@ -2745,20 +2757,15 @@ def get_all_disks_did(session, partition=False):
     :return: a dict with all disks did lists each include disk
              kname, serial and wwn.
     """
-    disks = list_linux_guest_disks(session, partition)
-    logging.debug("Disks detail: %s" % disks)
-    all_disks_did = {}
-    for line in disks:
-        kname = line.split('/')[2]
-        get_disk_info_cmd = "udevadm info -q property -p /sys/block/%s" % kname
-        output = session.cmd_output_safe(get_disk_info_cmd)
-        re_str = r"(?<=DEVNAME=/dev/)(.*)|(?<=ID_SERIAL=)(.*)|"
-        re_str += "(?<=ID_SERIAL_SHORT=)(.*)|(?<=ID_WWN=)(.*)"
-        did_list_group = re.finditer(re_str, output, re.M)
-        did_list = [match.group() for match in did_list_group if match]
-        all_disks_did[kname] = did_list
-
-    return all_disks_did
+    disks_did = get_linux_disks(session, parted)
+    for disk in disks_did.keys():
+        get_info_cmd = "udevadm info -q property /dev/%s" % disk
```

Comment:
On rhel6, should use cmd "udevadm info -q property -p /sys/block/sda" or "udevadm info -q property -n /dev/sda"